.github/scripts/check-anchors.py

```
@@ -0,0 +1,370 @@
+
+def git_changed_files(root: str) -> list:
+    """Files changed vs origin/main (fallback: HEAD~1 for direct pushes).
+
+    Returns both sides of a rename (pre- and post-image paths) so inbound
+    links to the old path are still validated after a file move.
+
+    A successful diff with no changes returns an empty list (no fallback);
+    the HEAD~1 fallback only runs when origin/main is unavailable. Raises
+    RuntimeError if both diffs fail.
+    """
+    origin_result = subprocess.run(
```

Ruff (0.16.1)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)